### ISSUE TITLE
deserialise json containing oneToMany relation with requery-jackson

### ISSUE THREAD

**Issue by bobthekingofegypt** [requery/requery#504]
I've been trying to deserialise a json object using requery-jackson and insert it with it's relation into a db.  I'm running into trouble but I'm not sure if I am just misunderstanding what requery-jackson does.

I'll try create a simple example.  I have 2 model object, LevelOne and LevelTwo with a onetomany relationship.
```
@Entity
@JsonDeserialize(as=LevelOneEntity.class)
public abstract class LevelOne implements Persistable {
	@Key
	long id;
	@OneToMany(cascade = {CascadeAction.DELETE, CascadeAction.SAVE})
	List<LevelTwo> levelTwos;
	
	public long getId() {
		return id;
	}
	public void setId(long id) {
		this.id = id;
	}
	public List<LevelTwo> getLevelTwos() {
		return levelTwos;
	}
	public void setLevelTwos(List<LevelTwo> levelTwos) {
		this.levelTwos = levelTwos;
	}
}
```

```
@Entity
@JsonDeserialize(as=LevelTwoEntity.class)
public abstract class LevelTwo implements Persistable {
	@Key
    long id;
    @ManyToOne
    LevelOne levelOne;
    
    public long getId() {
		return id;
	}
	public void setId(long id) {
		this.id = id;
	}
	public LevelOne getLevelOne() {
		return levelOne;
	}
	public void setLevelOne(LevelOne levelOne) {
		this.levelOne = levelOne;
	}
}
```

If I create the objects manually and call upsert/insert it works fine

```
LevelOne one = new LevelOneEntity();
one.setId(1l);
        
LevelTwo two1 = new LevelTwoEntity();
two1.setId(23l);
two1.setLevelOne(one);

LevelTwo two2 = new LevelTwoEntity();
two2.setId(25l);
two2.setLevelOne(one);
        
one.getLevelTwos().add(two1);
one.getLevelTwos().add(two2);
        
store.upsert(one);
```

But if I try using requery-jackson it doesn't populate the backreference or insert the relations.
```
        ObjectMapper mapper = new EntityMapper(Models.DEFAULT, store);
        mapper.setPropertyNamingStrategy(PropertyNamingStrategy.SNAKE_CASE);
        
        URL jsonFileUrl = Resources.getResource("sample.json");
        String json = Resources.toString(jsonFileUrl, Charsets.UTF_8);
        
        LevelOne levelOne = mapper.readValue(json, LevelOne.class);
        store.upsert(levelOne);
```

The json being
```
{
	"id": 1,
	"level_twos": [
		{
			"id": 1000
		},
		{
			"id": 1001
		}
	]
}
```

And requery initialised like this each time
```
        SQLiteDataSource dataSource = new SQLiteDataSource();
        dataSource.setUrl("jdbc:sqlite:test.sqlite");
        SQLiteConfig config = new SQLiteConfig();
        config.setDateClass("TEXT");
        dataSource.setConfig(config);
        EntityModel model = Models.DEFAULT;
        
        new SchemaModifier(dataSource, model).createTables(TableCreationMode.DROP_CREATE);
        io.requery.sql.Configuration configuration = new ConfigurationBuilder(dataSource, model)
                .useDefaultLogging()
                .build();
        
        EntityDataStore<Persistable> store = new EntityDataStore<Persistable>(configuration);
```

Is this something that should work or have I misunderstood the integration with jackson?

**Comment by npurushe** [comment]
That looks like it should work, but it also looks like your `level_twos` array contains only ids? will look into it.

**Comment by npurushe** [comment]
Should be resolved with the change in #563
